boring/src/x509/mod.rs

```diff
@@ -493,18 +493,18 @@ impl X509Ref {
     /// Returns the certificate's Not After validity period.
     pub fn not_after(&self) -> &Asn1TimeRef {
         unsafe {
-            let date = X509_getm_notAfter(self.as_ptr());
+            let date = X509_get0_notAfter(self.as_ptr());
```

Comment:
> since we're returning an unmutable reference, it's not possible to call any mutable functions of Asn1TimeRef (without using unsafe again): https://docs.rs/boring/1.1.6/boring/asn1/struct.Asn1TimeRef.html Another way to say it: returning an unmutable reference is effectively propagating the same guarantee as *const. (right?)

Doesn't an immutable reference guarantee that it won't be changed by this code *or any other code*? The docs say

> X509_getm_notBefore() and X509_getm_notAfter() are similar to X509_get0_notBefore() and X509_get0_notAfter() except they return non-constant mutable references

which seems to imply that these could change because the OpenSSL library itself decides to change them. I don't know if that's valid while another piece of code has an immutable reference to them.

Reply:
I think I was confused before - those docs are talking about `X509_getm_notAfter`, not `X509_get0_notAfter`. I think get0 should be ok as you're written it.